pkg/instrumental_adaptor/provider/translator.go

```diff
@@ -53,14 +53,12 @@ func (t *Translator) GetRespForExternalMetric(response *instrumental.Instrumenta
 
 func getExternalMetricName(metricName string) string {
 	re := regexp.MustCompile(`\|[a-z]?`)
-	var out string
 	ak := re.FindAllStringSubmatch(metricName, -1)
 	for _, v := range ak {
-		n := strings.Replace(v[0], "|", "", -1)
-		metricName = strings.Replace(metricName, n, strings.ToUpper(n), -1)
```

Comment:
the -1 here will replace every instance of the string n (which at this point in the original code would have been a single letter character. thus causing the example I showed earlier. but the FindAllStringSubmatch will find each instance that matches the regex even if they are duplicates.